opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -743,6 +781,100 @@ def set_up_accelerate_config(
         with open(file_path, "w") as file:
             yaml.dump(default_file, file)
 
+    def make_model_config_json(
+        self,
+        model_name: str = None,
+        version_number: int = 1,
+        embedding_dimension: int = 768,
+        all_config: str = None,
+        model_type: str = None,
+    ) -> None:
+        """
+        parse from config.json file of pre-trained hugging-face model to generate a model_config.json file. If all required
+        fields are given by users, use the given parameters and will skip reading the config.json,
+
+        Parameters
+        ----------
+        model_name: str = None
+            Optional, The name of the model. If None, default to parse from model id, for example,
+            'msmarco-distilbert-base-tas-b'
+        version_number: int = 1,
+            Optional, The version number of the model. If None, default to be 1.
+        embedding_dimension: int = 768,
+            Optional, the embedding_dimension of the model. If None, parse embedding_dimension from the config file of
+             pre-trained hugging-face model, if not found, default to be 768.
+        all_config: dict = None,
+            Optional, the all_config of the model. If None, parse all contents from the config file of pre-trained
+            hugging-face model.
+        model_type: str = None,
+            Optional, the model_type of the model. If None, parse model_type from the config file of pre-trained
+            hugging-face model.
+
+        Returns
+        -------
+        None
+        """
+        folder_path = self.folder_path
+        config_json_file_path = os.path.join(folder_path, "config.json")
+        if model_name is None:
+            model_name = self.model_id.split("/")[-1]
+
+        if all_config is None or model_type is None or embedding_dimension == 768:
```

Comment:
There are several branches here , we should add unit test to cover all. For example
Case1: all_config is None, model_type is None, embedding_dimension is 768
Case2: all_config is **not** None, model_type is **not** None, embedding_dimension is 768
Case3: ...
...

You can write test in another PR. Just remind we should cover all possible combinations

Reply:
that's good call. Given a second thought: when there  are user inputs for `model_type` and `embedding_dimension`, then we can skip reading the `config.json` file.

So here is the logic: 
IF model_type is None or embedding_dimension is None:
   check `config.json` exist, throw exceptions with file not found. 
  if exists reading file:
      if all_config is None:
                        load all_config
     if model_type is None:
                       load model_type
    if embedding_dimension is None:
                  find if embedding_dimension keys exist in file, 
                           if found, load embedding_dimension
                           else give default value 768. 
                           print message that didn't find embedding size in file and use default value. if there shall be a different value, suggest user to input in the argument.